Intersect.Server/Entities/Combat/Dash.cs

```diff
@@ -50,40 +51,60 @@ public partial class Dash
             bool zdimensionPass = false
         )
         {
-            var n = 0;
             en.MoveTimer = 0;
             Range = 0;
             for (var i = 1; i <= range; i++)
             {
-                n = en.CanMove(Direction);
-                if (n == -5) //Check for out of bounds
+                if (!en.CanMoveInDirection(Direction, out var blockerType, out var entityType))
                 {
-                    return;
-                } //Check for blocks
-
-                if (n == -2 && blockPass == false)
-                {
-                    return;
-                } //Check for ZDimensionTiles
-
-                if (n == -3 && zdimensionPass == false)
-                {
-                    return;
-                } //Check for active resources
-
-                if (n == (int) EntityType.Resource && activeResourcePass == false)
-                {
-                    return;
-                } //Check for dead resources
-
-                if (n == (int) EntityType.Resource && deadResourcePass == false)
-                {
-                    return;
-                } //Check for players and solid events
-
-                if (n == (int) EntityType.Player || n == (int) EntityType.Event)
-                {
-                    return;
+                    switch (blockerType)
+                    {
+                        case MovementBlockerType.OutOfBounds:
+                            return;
+
+                        case MovementBlockerType.MapAttribute:
+                            if (blockPass)
+                            {
+                                break;
+                            }
+
+                            return;
+
+                        case MovementBlockerType.ZDimension:
+                            if (zdimensionPass)
+                            {
+                                break;
+                            }
+
+                            return;
+
+                        case MovementBlockerType.Entity:
+                            switch (entityType)
+                            {
+                                case EntityType.Resource when activeResourcePass == false:
+                                case EntityType.Resource when deadResourcePass == false:
```

Comment:
Make these into fall-throughs with an if (pass) { break; }

1. using pattern matching kills the jump table when compiled (the entire reason to use a switch)
2. when *Pass is true, it's caught by default now instead, which throws an exception when it should not be

Reply:
```csharp
                                    case EntityType.Resource:
                                    if (activeResourcePass || deadResourcePass)
                                    {
                                        break;
                                    }

                                    return;
```